libcxx/test/std/containers/container.adaptors/flat.set/flat.set.cons/deduct.pass.cpp

```diff
@@ -0,0 +1,317 @@
+    assert(std::ranges::equal(m, sorted_arr));
+  }
+  {
+    std::flat_set<int> mo;
+    std::flat_set m(mo.begin(), mo.end(), C());
+    ASSERT_SAME_TYPE(decltype(m), std::flat_set<int, C>);
+  }
+  {
+    std::flat_set<int> mo;
+    std::flat_set m(mo.cbegin(), mo.cend(), C());
+    ASSERT_SAME_TYPE(decltype(m), std::flat_set<int, C>);
+  }
+}
+
+void test_initializer_list() {
+  const int sorted_arr[] = {1, 2, 3, INT_MAX};
+  {
+    std::flat_set m{1, 2, 1, INT_MAX, 3};
+
+    ASSERT_SAME_TYPE(decltype(m), std::flat_set<int>);
+    assert(std::ranges::equal(m, sorted_arr));
+  }
+  {
+    std::flat_set m(std::sorted_unique, {1, 2, 3, INT_MAX});
+
+    ASSERT_SAME_TYPE(decltype(m), std::flat_set<int>);
+    assert(std::ranges::equal(m, sorted_arr));
+  }
+  {
+    std::flat_set s = {1};
+    ASSERT_SAME_TYPE(decltype(s), std::flat_set<int>);
+    assert(s.size() == 1);
+  }
+  {
+    using M = std::flat_set<int>;
+    M m;
+    std::flat_set s{m, m}; // flat_set(initializer_list<M>)
+    ASSERT_SAME_TYPE(decltype(s), std::flat_set<M>);
+    assert(s.size() == 1);
+  }
+}
+
+void test_initializer_list_compare() {
+  const int sorted_arr[] = {INT_MAX, 3, 2, 1};
+  using C                = std::greater<long>;
+  {
+    std::flat_set m({1, 2, 1, INT_MAX, 3}, C());
+
+    ASSERT_SAME_TYPE(decltype(m), std::flat_set<int, C>);
+    assert(std::ranges::equal(m, sorted_arr));
+  }
+  {
+    std::flat_set m(std::sorted_unique, {INT_MAX, 3, 2, 1}, C());
+
+    ASSERT_SAME_TYPE(decltype(m), std::flat_set<int, C>);
+    assert(std::ranges::equal(m, sorted_arr));
+  }
+}
+
+void test_from_range() {
+  std::list<int> r     = {1, 2, 1, INT_MAX, 3};
+  const int expected[] = {1, 2, 3, INT_MAX};
+  {
+    std::flat_set s(std::from_range, r);
+    ASSERT_SAME_TYPE(decltype(s), std::flat_set<int, std::less<int>>);
+    assert(std::ranges::equal(s, expected));
+  }
+  {
+    std::flat_set s(std::from_range, r, test_allocator<long>(0, 42));
+    ASSERT_SAME_TYPE(decltype(s), std::flat_set<int, std::less<int>, std::vector<int, test_allocator<int>>>);
+    assert(std::ranges::equal(s, expected));
+    assert(std::move(s).extract().get_allocator().get_id() == 42);
+  }
+}
+
+void test_from_range_compare() {
+  std::list<int> r     = {1, 2, 1, INT_MAX, 3};
+  const int expected[] = {INT_MAX, 3, 2, 1};
+  {
+    std::flat_set s(std::from_range, r, std::greater<int>());
+    ASSERT_SAME_TYPE(decltype(s), std::flat_set<int, std::greater<int>>);
+    assert(std::ranges::equal(s, expected));
+  }
+  {
+    std::flat_set s(std::from_range, r, std::greater<int>(), test_allocator<long>(0, 42));
+    ASSERT_SAME_TYPE(decltype(s), std::flat_set<int, std::greater<int>, std::vector<int, test_allocator<int>>>);
+    assert(std::ranges::equal(s, expected));
+    assert(std::move(s).extract().get_allocator().get_id() == 42);
+  }
+}
+
+void test() {
+  // Each test function also tests the sorted_unique-prefixed and allocator-suffixed overloads.
```

Comment:
I would find it easier to understand if the exact deduction guides being tested were written down in a comment before each thing testing it.

At that point I would probably also inline these functions since the comment will provide more context than the terse function name.